src/imitation/algorithms/bc.py

```diff
@@ -198,85 +183,69 @@ def train(
         batch_num = 0
         for epoch_num in trange(n_epochs, desc="BC epoch"):
             while samples_so_far < (epoch_num + 1) * self.expert_dataset.size():
+                BaseAlgorithm._update_current_progress_remaining(
+                    self, batch_num, total_batches
+                )
+                BaseAlgorithm._update_learning_rate(self, self.policy.optimizer)
                 batch_num += 1
                 batch_dict = self.expert_dataset.sample(self.batch_size)
                 assert len(batch_dict["obs"]) == self.batch_size
                 samples_so_far += self.batch_size
-                feed_dict = {
-                    self._true_acts_ph: batch_dict["acts"],
-                    self.policy.obs_ph: batch_dict["obs"],
-                }
-                _, stats_dict = self.sess.run(
-                    [self._train_op, self._stats_dict], feed_dict=feed_dict
-                )
+
+                obs_tensor = th.as_tensor(batch_dict["obs"]).to(self.policy.device)
+                acts_tensor = th.as_tensor(batch_dict["acts"]).to(self.policy.device)
+                loss, stats_dict = self._calculate_loss(obs_tensor, acts_tensor)
+
+                self.policy.optimizer.zero_grad()
+                loss.backward()
+                self.policy.optimizer.step()
                 stats_dict["epoch_num"] = epoch_num
                 stats_dict["n_updates"] = batch_num
                 stats_dict["batch_size"] = len(batch_dict["obs"])
 
                 if batch_num % log_interval == 0:
                     for k, v in stats_dict.items():
-                        logger.logkv(k, v)
-                    logger.dumpkvs()
-                batch_num += 1
+                        logger.record(k, v)
+                    logger.dump()
 
             if on_epoch_end is not None:
                 on_epoch_end(locals())
 
-    def save_policy(self, policy_path: str):
-        """Save a policy to a pickle. Can be reloaded by `.reconstruct_policy()`.
+    def save_policy(self, policy_path: str) -> None:
+        """Save policy to a path. Can be reloaded by `.reconstruct_policy()`.
 
         Args:
             policy_path: path to save policy to.
         """
-        policy_params = self.sess.run(self.policy_variables)
-        data = {
-            "class": self.policy_class,
-            "kwargs": self.policy_kwargs,
-            "params": policy_params,
-        }
-        dirname = os.path.dirname(policy_path)
-        if dirname:
-            os.makedirs(dirname, exist_ok=True)
-        with open(policy_path, "wb") as fp:
-            cloudpickle.dump(data, fp)
+        self.policy.save(policy_path)
+
+    def load_policy(self, *args, **kwargs) -> None:
+        """Load policy that `.save_policy()` previously saved.
+
+        The `args` and `kwargs` are passed to `.reconstruct_policy()`.
+        """
+        self.policy = self.reconstruct_policy(*args, **kwargs)
 
     @staticmethod
     def reconstruct_policy(
```

Comment:
Our [style guide](https://google.github.io/styleguide/pyguide.html#217-function-and-method-decorators) recommends avoiding `@staticmethod` (you can just make it a module function, since it does not depend on state in any way).

Reply:
@qxcv Yeah it is just a `th.load()` now. I'll defer to you and others about removing it!

Reply:
@AdamGleave Noted! It's now a module function.